core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -202,8 +209,17 @@ case class TaskCommitDenied(jobID: Int, partitionID: Int, attemptID: Int) extend
  * the task crashed the JVM.
  */
 @DeveloperApi
-case class ExecutorLostFailure(execId: String) extends TaskFailedReason {
-  override def toErrorString: String = s"ExecutorLostFailure (executor ${execId} lost)"
+case class ExecutorLostFailure(execId: String, isNormalExit: Boolean = false)
+  extends TaskFailedReason {
+  override def toErrorString: String = {
+    if (!isNormalExit) {
+      s"ExecutorLostFailure (executor ${execId} exited abnormally)"
+    } else {
+      s"ExecutorLostFailure (executor $execId} exited normally)"
+    }
+  }
+
```

Comment:
Ah, internal types that are serialized in event logs... :-/ The usual approach is to make new fields `Option[...]`, but that seems like overkill here. If you fixed JsonProtocolSuite then it's good enough for me.